e2e/runner/upgrade.go

```diff
@@ -105,18 +105,42 @@ func (r *E2ERunner) AssertAfterUpgrade(assertVersion string, assertFunc func())
 	oldVersion := fmt.Sprintf("v%s", os.Getenv("OLD_VERSION"))
 
 	// run these assertions only on the second run of the upgrade
-	if !r.IsRunningUpgrade() || !versionMajorIsZero || assertVersion != oldVersion {
+	if !r.IsRunningUpgrade() || !versionMajorIsZero || checkVersion(assertVersion, oldVersion) {
 		return
 	}
 	r.Logger.Print("🏃 Running assertions after upgrade for version: %s", assertVersion)
 	assertFunc()
 }
 
-func (r *E2ERunner) AssertBeforeUpgrade(assertVersion string, assertFunc func()) {
+// AddZetaE2EPreUpgradeHandler adds a handler to run any logic before an upgrade
+func (r *E2ERunner) AddZetaE2EPreUpgradeHandler(upgradeFrom string, preHandler func()) {
+	currentVersion := r.GetZetacoredVersion()
 	// run these assertions only on the first run of the upgrade
-	if !r.IsRunningUpgrade() || assertVersion != r.GetZetacoredVersion() {
+	if !r.IsRunningUpgrade() || checkVersion(upgradeFrom, currentVersion) {
 		return
 	}
-	r.Logger.Print("🏃 Running assertions before upgrade for version: %s", assertVersion)
-	assertFunc()
+	r.Logger.Print("🏃 Running assertions before upgrade for version: %s", upgradeFrom)
+	preHandler()
+}
+
+// AddZetaE2EPostUpgradeHandler adds a handler to run any logic after and upgrade to enable tests to be executed
+// Note This is handler is not related to the cosmos-sdk upgrade handler in any way
+func (r *E2ERunner) AddZetaE2EPostUpgradeHandler(upgradeFrom string, postHandler func()) {
```

Comment:
```suggestion
func (r *E2ERunner) AddPostUpgradeHandler(upgradeFrom string, postHandler func()) {
```
Seems we make the name unnecessarily long here